x-pack/plugins/watcher/public/application/models/visualize_options/visualize_options.js

```diff
@@ -6,19 +6,29 @@
  */
 
 export class VisualizeOptions {
-  constructor(props = {}) {
+  constructor(props = {}, kibanaVersion) {
     this.rangeFrom = props.rangeFrom;
     this.rangeTo = props.rangeTo;
     this.interval = props.interval;
     this.timezone = props.timezone;
+    this.kibanaVersion = kibanaVersion;
   }
 
   get upstreamJson() {
-    return {
+    const json = {
       rangeFrom: this.rangeFrom,
       rangeTo: this.rangeTo,
-      interval: this.interval,
       timezone: this.timezone,
     };
+
+    if (this.kibanaVersion < 8) {
+      // In 7.x we still use the deprecated "interval" option
+      json.interval = this.interval;
+    } else {
+      // From 8.x we use the more precise "fixed_interval"
+      json.fixed_interval = this.interval;
```

Comment:
If we apply my suggestion to delay differentiation to the last opportunity, we can also revert these changes to the client.

Reply:
I initially wanted to use the injected kibana version to not rely on a constant. But we anyway need the constant for the test **and** the change you suggest to do above. So I'll get rid of this change and we'll rely on the constant.